bin/nimbus-fml.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Nimbus Feature Manifest Language Generator
+#
+# For more infomration, check out https://experimenter.info/fml-spec
+#
+# This script generates Swift definitions for all the experimentable features supported by Nimbus. 
+# It generates Swift code to be included in the final build.
+#
+# To use it in a Swift project, follow these steps:
+# 1. Import the `nimbus-fml.sh` script into your project.
+# 2. Add a `<NAME>.yaml` feature manifest file. Check out https://experimenter.info/fml-spec for the spec.
+# 3. Add a new "Run Script" build step and set the command to `bash $PWD/nimbus-fml.sh`
+# 4. Add your FML filee as an Input File for the "Run Script" step.
+# 5. Run the build.
+# 6. Add the "FML.swift" file in the `Generated` folder to your project.
+# 7. Add the same "FML.swift" from the `Generated` folder as Output Files of the newly created "Run SCript" step.
+# 8. Start using the generated feature code.
+
+set -e
+
+# CMDNAME is used in the usage text below.
+# shellcheck disable=SC2034
+CMDNAME=$(basename "$0")
+USAGE=$(cat <<'HEREDOC'
+$(CMDNAME)
+Tarik Eshaq <teshaq@mozilla.com>
+
+Nimbus Feature Manifest Language generator.
+
+This script generates the code needed to interact with Nimbus, exposing features which are experimentable.
+
+For more infomration, check out https://experimenter.info/fml-spec
+
+The script structure was adopted from the similar script "sdk_generator.sh" written by the Glean team.
+
+This script should be executed as a "Run Build Script" phase from Xcode.
+
+USAGE:
+    ${CMDNAME} [OPTIONS] <FML_PATH>
+
+ARGS:
+    <FML_PATH>  The path to the FML yaml file to parse. default: \$SCRIPT_INPUT_FILE_0 environment variable.
+
+OPTIONS:
+    -o, --output  <PATH>             Folder to place generated FML code in. Default: \$SOURCE_ROOT/\$PROJECT/Generated
+    -h, --help                       Display this help message.
+HEREDOC
+)
+
+helptext() {
+    echo "$USAGE"
+}
+
+FML_PATH=
+OUTPUT_DIR="${SOURCE_ROOT}/${PROJECT}/Generated"
+AS_VERSION=v91.0.0
```

Comment:
Hmm yeah - this will be annoying to keep track of (and we will 100% forget to upgrade it if we don't introduce breaking changes.. we can have the script try to extract the version out of the xcode project, but that will be fragile 😕